modules/polaris/src/main/ts/ephox/polaris/words/Words.ts

```diff
@@ -51,12 +53,13 @@ const findWordsWithIndices = <T>(chars: Word<T>, sChars: string[], characterMap:
     word.push(chars[i]);
 
     // If there's a word boundary between the current character and the next character,
+    // (and this boundary doesn't depend from a dot at the end of an acronym)
     // append the current word to the words array and start building a new word.
-    if (isWordBoundary(characterMap, i)) {
+    if (isWordBoundary(characterMap, i) && !isAnAcronym(word.join('') + sChars[i + 1])) {
       const ch = sChars[i];
       if (
         (options.includeWhitespace || !WHITESPACE.test(ch)) &&
-        (options.includePunctuation || !PUNCTUATION.test(ch))
+        (options.includePunctuation || !PUNCTUATION.test(ch) || isAnAcronym(word.join('')))
```

Comment:
I wonder if this could cause performance issues on larger documents? This is also used for word count for example.

Reply:
the problem here as that I did not find any alternative, I putted the check at the end and in the lowest level to make it be triggered only on boundaries and if we are checking a `dot`